packages/apollo-gateway/src/index.ts

```diff
@@ -82,6 +126,16 @@ export class ApolloGateway implements GraphQLService {
   private engineConfig: GraphQLServiceEngineConfig | undefined;
   private pollingTimer?: NodeJS.Timer;
   private onSchemaChangeListeners = new Set<SchemaChangeCallback>();
+  private serviceDefinitions: ServiceDefinition[] = [];
+
+  // Observe query plan, service info, and operation info prior to execution. The information made available here will give insight into the resulting query plan and the inputs that generated it.
+  protected experimental_didResolveQueryPlan?: DidResolveQueryPlanCallback;
+  // Observe composition failures and the ServiceList that caused them. Pretty straightforward, this enables reporting any issues that occur during composition. Implementors will be interested in addressing these immediately.
```

Comment:
nit: could we remove the pretty straightforward bit? I don't want to make people feel bad if they don't understand.

Reply:
I like the idea of strongly recommending that implementors tie this into their alerting! If only these were less experimental and we could tie them into public docs

Reply:
Copy-pasta got me on the wording 😆 I definitely think we could be more heavy-handed with recommendations once this isn't `experimental_`